scalameta/scalameta/shared/src/test/scala/scala/meta/tests/prettyprinters/PublicSuite.scala

```diff
@@ -133,10 +139,10 @@ class PublicSuite extends FunSuite {
   test("scala.meta.inputs.Input.File.toString") {
     import java.io._
     import java.nio.charset.Charset
-    val file = new File("hello.scala")
-    val input1 = Input.File(file, Charset.forName("cp1251"))
```

Comment:
Technically, it is *complete* in the sense that it supports all [Standard Charsets](https://docs.oracle.com/javase/8/docs/api/java/nio/charset/Charset.html), as defined by the documentation, which are the only one required to be implemented. It doesn't provide any *other* charset but the standard ones, though, so indeed cp1251 is not supported.